pkg/config/proxyrule/rule.go

```diff
@@ -197,6 +201,20 @@ type PostFilter struct {
 	CheckPermissionTemplate *StringOrTemplate `json:"checkPermissionTemplate" validate:"required"`
 }
 
+// WatchConfig defines a watch request to perform if the request matches.
+type WatchConfig struct {
```

Comment:
I'm having trouble finding where this is actually used - am I just missing it? I think this is covered by the prefilter rules